### ISSUE TITLE
Support for Tesla via HA Custom Integration 

### ISSUE THREAD

**Issue by CommanderROR9** [jonasbkarlsson/ev_smart_charging#86]
I just discovered your HACS Integration and it fits well with what I am currently building.
I currently still have static electricity prices, but am thinking about going with Tibber in the future. To test things first I am currently getting prices from Nordpool integration.

Charging my PV battery with cheap electricity is something I have already integrated, so my Tesla is the next step. Since there is a well maintained Tesla integration within HA this should be easy to add, I guess.
One important factor for me is deciding between Solar Charging and cheap Grid Charging. I hope that is something that would he configurable within your Integration. For the PV battery I use the Solcast Forecast to determine how much to charge from Grid.

**Comment by jonasbkarlsson** [comment]
As of now, this integration only optimize the EV charging based on the electricity prices. There is no coordination with charging other batteries or using solar power for charging. In the near future, there will be a v1.5.0 available that will contain the possibility to control all the steering parameters of the integration with automations (using service calls). Maybe that will make this integration more interesting for you.

**Comment by CommanderROR9** [comment]
Thank you for your reply!
Version 1.5 sounds very interesting...
The current version is already very useful, and I could probably build a
pretty simple automation that switches between cheap grid Charging and
Solar Charging depending on a few parameters.

That might actually be better than putting both into one Integration. At
the end of the day, having a "Central Energy Manager" that coordinates
everything would be the holy grail, but it might be too inflexible and
complex.

So for now I am thinking about something like this:
I have a "Solar Forecast" integration that tries to guesstimate how much
Solar Yield I will have today and tomorrow. I also have a "Solar Charge"
Automation for my Tesla that I built based on some preexisting code in the
HA Forum. I can turn the "Solar Charge" related automations on an off with
a virtual switch, and I can also turn the EV Smart Charge on an off with a
virtual switch, so if I created an Automation that monitors how much Solar
to expect for the next day, that could then turn the Solar Charge on for
days with more than 15kWh of expected Yield and turn the EV Smart Charge on
for days with less than 10kWh of Solar...or something along those lines. Of
course I would have to take my home battery into account as well, but
that's not too complicated I guess.
My main "worry" right now is that I might be charging my PV Battery from
the Grid and then moving that electricity into my Tesla which would be very
stupid. I still have to figure out how to prevent that ..

Anyway, Happy New Year and keep up the good work!

Jonas Karlsson ***@***.***> schrieb am Sa., 31. Dez. 2022,
15:36:
